packages/jest-snapshot/src/Snapshot.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const fs = require('fs');
+
+class Snapshot {
+
+  constructor(filename) {
+    this._filename = filename;
+    this._loaded = null;
+  }
+
+  _getLoaded() {
```

Comment:
can you get rid of this and just call the function below `read` (to be in line with `jest-haste-map`). Why can't load do the caching check? We only need to load each file once, so it can just live there.